components/common/BlobListLightbox.tsx

```diff
@@ -0,0 +1,89 @@
+import { useServerConfig } from '@/shell/ConfigurationContext'
+import { ToolsOzoneModerationDefs } from '@atproto/api'
+import { useCallback } from 'react'
+import Lightbox from 'yet-another-react-lightbox'
+import Captions from 'yet-another-react-lightbox/plugins/captions'
+import Thumbnails from 'yet-another-react-lightbox/plugins/thumbnails'
+
+export const useBlobUrl = ({ authorDid }: { authorDid: string }) => {
+  const { appview } = useServerConfig()
+  const cdnUrl = appview?.endsWith('.bsky.app')
+    ? 'https://cdn.bsky.app'
+    : appview
+
+  return useCallback(
+    ({
+      cid,
+      isAvatar,
+      size = 'fullsize',
+    }: {
+      cid: string
+      isAvatar?: boolean
+      size?: 'thumbnail' | 'fullsize'
+    }) => {
+      let sourcePath = isAvatar ? 'avatar' : 'feed'
+
+      // avatar_fullsize doesn't exist, instead the default avatar/ serves the full size image
+      if (!isAvatar && size !== 'fullsize') {
+        sourcePath += `_${size}`
+      }
+
+      return `${cdnUrl}/img/feed_${size}/plain/${authorDid}/${cid}@jpeg`
+    },
+    [authorDid, cdnUrl],
+  )
+}
+
+export const BlobListLightbox = ({
+  authorDid,
+  blobs,
+  onClose,
+  slideIndex,
+}: {
+  authorDid: string
+  blobs: ToolsOzoneModerationDefs.BlobView[]
+  onClose: () => void
+  slideIndex: number
+}) => {
+  const getBlobUrl = useBlobUrl({ authorDid })
+  const handleKeyDown = (event) => {
+    if (event.key === 'Escape') {
+      event.stopPropagation()
+    }
+  }
```

Comment:
You need to `useCallback` here. Otherwise the `document.addEventListener('keydown', handleKeyDown)` and `document.removeEventListener('keydown', handleKeyDown)` might not refer to the same function (resulting in the listener staying there forever)

Reply:
ermm.. not sure if that's correct? what would the deps for the hook be?

Reply:
happy to address this later and if we find this to be an issue, this will need to be updated in a few other places where lightbox is used.